spark-janelia

```diff
@@ -145,17 +143,94 @@ elif args.task == 'start-scala':
     os.environ['MASTER'] = f.readline().replace('\n','')
     os.system(version + '/bin/spark-shell')
 
-elif args.task == 'submit':
 
+def submit(master = ''):
     os.environ['SPARK_HOME'] = version
 
     if os.getenv('PATH') is None:
         os.environ['PATH'] = ""
 
     os.environ['PATH'] = os.environ['PATH'] + ":" + "/usr/local/python-2.7.6/bin"
-    f = open(os.path.expanduser("~") + '/spark-master', 'r')
-    master = f.readline().replace('\n','')
+    if master == '':
+        with open(os.path.expanduser("~") + '/spark-master', 'r') as f:
+            master = f.readline().replace('\n','')
     os.environ['MASTER'] = master
     os.system(version + '/bin/spark-submit --master ' + master + ' ' + args.submitargs)
 
 
+def submitAndDestroy( master, jobId ):
+    submit(master)
+    destroy(jobId)
+
+if __name__ == "__main__":
+
+    parser = argparse.ArgumentParser(description="launch and manage spark cluster jobs")
+                        
+    choices = ('launch', 'login', 'destroy', 'start', 'start-scala', 'submit', 'lsd')
+                        
+    parser.add_argument("task", choices=choices)
+    parser.add_argument("-n", "--nnodes", type=int, default=2, required=False)
+    parser.add_argument("-i", "--ipython", action="store_true")
+    parser.add_argument("-v", "--version", choices=("1", "2", "3"), default="1", required=False)
+    parser.add_argument("-j", "--jobid", type=int, default=None, required=False)
+    parser.add_argument("-t", "--sleep_time", type=int, default=86400, required=False)
+    parser.add_argument("-s", "--submitargs", type=str, default='', required=False)
+                        
+    args = parser.parse_args()
+                        
+    SPARKVERSIONS = {   
+        '1': '/usr/local/spark-current',
+        '2': '/usr/local/spark-master',
+        '3': '/usr/local/spark-test'
+    }                   
+                        
+    SPARKVERSIONSSHORT = {
+        '1': 'spark',   
+        '2': 'spark2',  
+        '3': 'spark3'   
+    }                   
+                        
+    version = SPARKVERSIONS[args.version]
+    version_short = SPARKVERSIONSSHORT[args.version]
+                        
+    if args.nnodes == 1:
+        raise Exception('Cannot start a Spark cluster with only 1 node, please request 2 or more.')
+                        
+    if args.task == 'launch':
+        launch(str(args.sleep_time))
+                        
+    elif args.task == 'login':
+        login()         
+                        
+    elif args.task == 'destroy':
+        destroy(args.jobid or '')
+                        
+                        
+    elif args.task == 'start':
+        start()         
+                        
+                        
+    elif args.task == 'start-scala':
+        start_scala()   
```

Comment:
True, we should stick to one style. Let's go with whatever your preference is. I will modify the stuff accordingly.